ipfix/ipfix.c

```diff
@@ -140,14 +143,79 @@ flow_meter_enable_disable_command_fn (vlib_main_t * vm,
   return 0;
 }
 
+static clib_error_t * ipfix_set_command_fn (vlib_main_t * vm,
+                                            unformat_input_t * input,
+                                            vlib_cli_command_t * cmd)
+{
+  u32 val = 0;
+  ip4_address_t addr;
+  ipfix_main_t * im = &ipfix_main;
+
+  while (unformat_check_input (input) != UNFORMAT_END_OF_INPUT) {
+    if (unformat(input, "timeout")) {
+      if (unformat(input, "idle %u", &val)) {
+        im->idle_flow_timeout = val * 1e3;
+      } else if (unformat(input, "active %u", &val)) {
+        im->active_flow_timeout = val * 1e3;
+      } else if (unformat(input, "template %u", &val)) {
+        im->template_timeout = val * 1e3;
+      } else {
+        return clib_error_return(0,
+                                 "expected timeout command, got `%U`",
+                                 format_unformat_error, input);
+      }
+    } else if (unformat(input, "port")) {
+      if (unformat(input, "exporter %u", &val)) {
+        if (val > 65536) {
+          return clib_error_return(0, "expected valid port");
+        }
+        im->exporter_port = val;
+      } else if (unformat(input, "collector %u")) {
+        if (val > 65536) {
+          return clib_error_return(0, "expected valid port");
+        }
+        im->collector_port = val;
+      } else {
+        return clib_error_return(0,
+                                 "expected port command, got `%U`",
+                                 format_unformat_error, input);
+      }
+    } else if (unformat(input, "ip")) {
+      if (unformat(input, "exporter %U", unformat_ip4_address, &addr)) {
+        im->exporter_ip = addr;
+      } else if (unformat(input, "collector %U", unformat_ip4_address, &addr)) {
+        im->collector_ip = addr;
+      } else {
+        return clib_error_return(0,
+                                 "expected port command, got `%U`",
+                                 format_unformat_error, input);
+      }
+    } else if (unformat(input, "observation-domain %u", &val)) {
+      im->observation_domain = val;
+    } else {
+      return clib_error_return(0, "unknown command");
+    }
+  }
+
+  return 0;
+}
+
 /**
  * @brief CLI command to enable/disable the ipfix plugin.
  */
-VLIB_CLI_COMMAND (sr_content_command, static) = {
-    .path = "ipfix flow-meter",
-    .short_help = 
-    "ipfix flow-meter <interface-name> [disable]",
-    .function = flow_meter_enable_disable_command_fn,
+VLIB_CLI_COMMAND (ipfix_enable_command, static) = {
+  .path = "ipfix flow-meter",
+  .short_help = "ipfix flow-meter <interface-name> [disable]",
+  .function = flow_meter_enable_disable_command_fn,
+};
+
+/**
+ * @brief CLI command to set options for the ipfix plugin.
+ */
+VLIB_CLI_COMMAND (ipfix_set_command, static) = {
+  .path = "set ipfix",
+  .short_help = "set ipfix timeout {idle|active|template} <seconds>",
```

Comment:
This help command should be extended to accept port collector, ip collector, etc. right?

Reply:
Thanks, fixed now.